### PR TITLE
Require Kokkos 4.3

### PR DESCRIPTION
**Pull request by aprokop**
- Require Kokkos 4.3
- Update CI using 4.2 to using 4.4

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: .jenkins/continuous.groovy [hunk 1]
```diff
@@ -102,7 +102,7 @@ pipeline {
                         dockerfile {
                             filename "Dockerfile"
                             dir "docker"
-                            additionalBuildArgs '--build-arg BASE=nvidia/cuda:11.1.1-devel-ubuntu20.04 --build-arg KOKKOS_OPTIONS="-DCMAKE_CXX_EXTENSIONS=OFF -DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON"'
+                            additionalBuildArgs '--build-arg BASE=nvidia/cuda:11.1.1-devel-ubuntu20.04 --build-arg KOKKOS_VERSION=4.4.00 ---build-arg KOKKOS_OPTIONS="-DCMAKE_CXX_EXTENSIONS=OFF -DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON"'
```
Comment (dalg24) [inline .jenkins/continuous.groovy:105]: Is it intentional that we are leapfrogging the Kokkos versions between all the builds?

Comment (aprokop) [reply .jenkins/continuous.groovy:105]: > Is it intentional that we are leapfrogging the Kokkos versions between all the builds?

Yes. We have about half builds on version $n$ and half on $n+1$. When updating for the next release, we switch $n$ build to $n+2$.

**Comment by aprokop** [comment]
CUDA-11.1.1 build failed with unrelated error
```
```

**Comment by aprokop** [comment]
GPU-aware CUDA build has one unrelated test failure.

**Pull request merged**